src/plugin.js

```diff
@@ -70,6 +75,9 @@ class WebpackRequireFrom {
         "Object.defineProperty(" + mainTemplate.requireFn + ', "p", {',
         "  get: function () {",
         getterBody,
+        " },",
+        "  set: function (value) {",
+        "",
```

Comment:
Curious why did you add a setter?

Reply:
Because webpack allow users to change the `__webpack_public_path__` in their code for a dynamic outputPath, if their is no setter in property `p`, the application will fail with a error message:
`Cannot set property p of function...`
See: 
https://webpack.js.org/guides/public-path/#on-the-fly
so i think it's better to add a setter to warn the user in the setter function like:
```
        "  set: function (value) {",
        `console.warn("You should not change __webpack_public_path__ directly, use ${PLUGIN_NAME} to config public path.")`,
        " }",
```

Reply:
Do you think it's necessary?

Reply:
@agoldis 
Yes, I also think it's better to handle this scenario in separate issue gently, a warning in setter function is just temporary in my case. I will remove this part from the PR. :)